File: solutions/c/collatz-conjecture/1/collatz_conjecture.c
#include "collatz_conjecture.h"
#include <complex.h>
#include <stdio.h>

// void main() {
//     steps(1000000);
// }

int steps(int n) {
    int steps = 0;

    if (n < 1) {
        return -1;
    };
    while (n > 1) {
        if (n % 2 == 0) {
            n /= 2;
        } else {
            n *= 3;
            n++;
        }
        steps++;
    }
    return steps;
}

File: solutions/go/collatz-conjecture/1/collatz_conjecture.go
package collatzconjecture

import "errors"

func CollatzConjecture(n int) (int, error) {
	if n < 1 {
		return -1, errors.New("Number is smaller than 1")
	}

	steps := 0
	for n != 1 {
		if n%2 == 0 {
			n /= 2
		} else {
			n *= 3
			n += 1
		}
		steps++
	}
	return steps, nil
}
